erdos/src/communication/control_plane/notifications.rs

```diff
@@ -0,0 +1,68 @@
+//! A collection of enums that specify the communication patterns between
+//! [`Leader`](crate::node::Leader), [`Worker`](crate::node::WorkerNode)
+//! and [`Driver`](crate::dataflow::graph::Job::Driver)s.
+use std::{collections::HashMap, net::SocketAddr};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    dataflow::graph::{AbstractJobGraph, Job, JobGraph, JobGraphId},
+    node::{JobState, Resources, WorkerId},
+};
+
+/// An enum specifying the notifications that a [`Driver`](crate::dataflow::graph::Job::Driver)
+/// can send to a [`Worker`](crate::node::WorkerNode).
```

Comment:
The idea was that a `Client` is just a thin-wrapper over a`Worker` with empty resources. This makes the implementation easier, since the `Leader` can now schedule jobs to this `Worker`, but the only schedulable Job is the `Driver` job and the `Worker` then uses the `DataPlane` abstraction to connect to the corresponding sources and sinks.

Reply:
Assuming s/sources/ingress\ streams and s/sinks/egress\ streams, that SGTM!